org.eclipse.scout.rt.mail/src/main/java/org/eclipse/scout/rt/mail/smtp/SmtpConnectionPool.java

```diff
@@ -158,22 +168,37 @@ public void sendMessage(SmtpServerConfig smtpServerConfig, MimeMessage message,
    */
   protected SmtpConnectionPoolEntry leaseConnection(SmtpServerConfig smtpServerConfig) {
     Assertions.assertGreater(smtpServerConfig.getPoolSize(), 0, "Pool size of provided SmtpServerConfig must be greater 0.");
-    synchronized (m_poolLock) {
-      Assertions.assertFalse(m_destroyed, "SmtpConnectionPool not available because it has already been destroyed.");
-      SmtpConnectionPoolEntry candidate = null;
-      while (candidate == null) {
+    while (true) {
```

Comment:
The only way we loop again is when there createNewConnection was false and we had to wait until one gets available, right? I.e. m_poolLock was notified.

Reply:
Yes, that is how it is supposed to work - added a comment in 231195a99e67f0e19d31639e53a5fd6b7b791174